src/Backup.ts

```diff
@@ -286,6 +285,35 @@ class Backup {
           await this.showError(i18n.__("msg.error.folderCreation", e.message));
         }
       }
+    };
+
+    if (this.createSubfolder) {
+      this.log.verbose("append subFolder");
+      this.backupBasePath = path.join(this.backupBasePath, "JoplinBackup");
+      await handleSubfolderCreation();
+    }
+
+    if (this.createSubfolderPerProfile) {
+      this.log.verbose("append profile subfolder");
+      // We assume that Joplin's profile structure is the following
+      //   rootProfileDir/
+      //   | profileDir/
+      //   | | [[profile content]]
+      // or, if using the default,
+      //   rootProfileDir/
+      //   | [[profile content]]

```

Comment:
This is what Joplin's profile structure currently looks like. However, it doesn't seem to be documented anywhere...

Reading `profiles.json` (which seems to not always exist) in the root profile directory may be another way to get this information. It also doesn't seem to be documented though.